### PR TITLE
feat: add sprig device mode integration

### PR DESCRIPTION
**Pull request by 1abhishekpandey**
# Description

- In this PR we added the Sprig mobile device mode integration.
- The following event types are supported: identify and track.
- RESET call is also supported.
- `Rudder-Sprig` will be distributed as both `CoaoaPod` and `SPM`, as v1.0.0.
- In order to display the survey, the `ViewController` instance must be passed before making a track event.
```Swift
RudderSprigFactory.instance.setViewController(self)
```
- The survey looks like this:
image

### CODE REVIEW

File: Screenshots/add_rudder_sprig.png
Comment (SKannaniOS) [inline Screenshots/add_rudder_sprig.png:1]: Suggestion: Ensure that the screenshot you’re adding to the README displays only the target package, excluding any other packages.

**Review by SKannaniOS — APPROVED** [review]

File: Screenshots/add_rudder_sprig.png
Comment (1abhishekpandey) [reply Screenshots/add_rudder_sprig.png:1]: Thanks, I've addressed this comment.

**Review by ChryssaAliferi — APPROVED** [review]
👍 Looks good to me

**Review by vgupta98 — APPROVED** [review]

**Comment by 1abhishekpandey** [comment]
Release have been made in this PR: https://github.com/rudderlabs/rudder-integration-sprig-ios/pull/6. Hence closing this.

**Pull request closed without merging**